dsl/src/lib.rs

```diff
@@ -0,0 +1,175 @@
+extern crate quote;
+extern crate litrs;
+extern crate proc_macro2;
+
+use quote::quote;
+
+use proc_macro2::TokenTree;
+use proc_macro2::TokenStream;
+use proc_macro2::TokenTree::Punct;
+use proc_macro2::TokenTree::Ident;
+use proc_macro2::token_stream::IntoIter;
+use std::iter::Peekable;
+
+use std::convert::TryFrom;
+use std::str::FromStr;
+use litrs::Literal;
+
+#[derive(PartialEq)]
+enum ExprType {
+    Nil, Int, Bool
+}
+
+// within the compiler, these generate tokens for
+// Value
+struct ParsedExpr {
+    t: ExprType,
+    tokens: TokenStream,
+}
+
+fn parse_literal(it: &mut Peekable<IntoIter>) -> ParsedExpr {
+    let token = it.next().expect("failed to parse literal, hit end of string");
+
+    match token {
+        Punct(ref x) => if x.as_char() == '(' {
+            let v = parse_expr(it);
+
+            it.next().expect("expected closing paren, hit end of string");
+            return v
+        },
+
+        // boolean literals
+        Ident(ref x) => match x.to_string().as_str() {
+            "true"  => return ParsedExpr{ t: ExprType::Bool, tokens: quote!{ true }},
+            "false" => return ParsedExpr{ t: ExprType::Bool, tokens: quote!{ false }},
+
+            // unary not
+            "not"   => {
```

Comment:
A bit of bikeshed: I think that a DSL used in Rust is better off using a syntax that resembles Rust. Thus, suggest using `!` to denote the "not" operation

Reply:
Why not both?